web/src/features/organizations/server/organizationRouter.ts

```diff
@@ -0,0 +1,101 @@
+import {
+  createTRPCRouter,
+  protectedOrganizationProcedure,
+  protectedProcedure,
+} from "@/src/server/api/trpc";
+import { auditLog } from "@/src/features/audit-logs/auditLog";
+import { organizationNameSchema } from "@/src/features/organizations/utils/organizationNameSchema";
+import * as z from "zod";
+import { throwIfNoOrganizationAccess } from "@/src/features/rbac/utils/checkOrganizationAccess";
+import { TRPCError } from "@trpc/server";
+
+export const organizationsRouter = createTRPCRouter({
+  create: protectedProcedure
+    .input(organizationNameSchema)
+    .mutation(async ({ input, ctx }) => {
+      if (!ctx.session.user.canCreateOrganizations)
+        throw new TRPCError({
+          code: "FORBIDDEN",
+          message: "You do not have permission to create organizations",
+        });
+
+      const organization = await ctx.prisma.organization.create({
+        data: {
+          name: input.name,
+          organizationMemberships: {
+            create: {
+              userId: ctx.session.user.id,
+              role: "OWNER",
+            },
+          },
+        },
+      });
+      await auditLog({
+        resourceType: "organization",
+        resourceId: organization.id,
+        action: "create",
+        orgId: organization.id,
+        orgRole: "OWNER",
+        userId: ctx.session.user.id,
+        after: organization,
+      });
+
+      return {
+        id: organization.id,
+        name: organization.name,
+        role: "OWNER",
+      };
+    }),
+  update: protectedOrganizationProcedure
+    .input(
+      organizationNameSchema.extend({
+        orgId: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      throwIfNoOrganizationAccess({
+        session: ctx.session,
+        organizationId: input.orgId,
+        scope: "organizations:update",
+      });
+      const beforeOrganization = await ctx.prisma.organization.findFirst({
+        where: {
+          id: input.orgId,
+          organizationMemberships: {
+            some: {
+              userId: ctx.session.user.id,
+              role: "OWNER",
```

Comment:
This can lead to unexpected behavior if we change permissions in throwIfNoOrganizationAccess. In the future, maybe different roles will be able to update.

Reply:
ty